runtime/src/lib.rs

```diff
@@ -158,6 +165,83 @@ pub fn native_version() -> NativeVersion {
 
 const NORMAL_DISPATCH_RATIO: Perbill = Perbill::from_percent(75);
 
+impl ChainExtension<Runtime> for ExampleExtension
+where
+	Runtime: SysConfig + pallet_contracts::Config,
+	<Runtime as SysConfig>::AccountId: UncheckedFrom<<Runtime as SysConfig>::Hash> + AsRef<[u8]>,
+{
+	fn call<E: Ext>(func_id: u32, env: Environment<E, InitState>) -> Result<RetVal, DispatchError>
+	where
+		E: Ext<T = Runtime>,
+		<E::T as SysConfig>::AccountId: UncheckedFrom<<E::T as SysConfig>::Hash> + AsRef<[u8]>,
+	{
+		// Match on function id assigned in the contract
+		match func_id {
+			// do_store_in_runtime
+			1 => {
+				let mut env = env.buf_in_buf_out();
+				let base_weight = RocksDbWeight::get().writes(1);
+				let overhead = <Runtime as pallet_contracts::Config>::Schedule::get()
+					.host_fn_weights
+					.debug_message;
+				env.charge_weight(base_weight.saturating_add(overhead))?;
+				// retrieve argument that was passed in smart contract invocation
+				let value: u32 = env.read_as()?;
+				let caller = env.ext().caller().clone();
+
+				crate::pallet_template::Pallet::<Runtime>::insert_number(
+					RawOrigin::Signed(caller).into(),
+					value,
+				)?;
+			},
+			// do_balance_transfer
+			2 => {
+				let mut env = env.buf_in_buf_out();
+				// Retrieve arguments
+				let (transfer_amount, recipient): (u32, AccountId) = env.read_as()?;
+
+				let base_weight = <Runtime as pallet_contracts::Config>::Schedule::get()
+					.host_fn_weights
+					.call_transfer_surcharge;
+				let overhead = <Runtime as pallet_contracts::Config>::Schedule::get()
+					.host_fn_weights
+					.debug_message;
+				env.charge_weight(base_weight.saturating_add(overhead))?;
+
+				let caller = env.ext().caller().clone();
+				let recipient_account = sp_runtime::MultiAddress::Id(recipient);
+
+				pallet_balances::Pallet::<Runtime>::transfer(
+					RawOrigin::Signed(caller).into(),
+					recipient_account,
+					transfer_amount.into(),
+				);
```

Comment:
Pretty sure this returns a result.
https://paritytech.github.io/substrate/master/pallet_balances/pallet/struct.Pallet.html#method.transfer

So you probably want to use `?`

Should've also given a warning at compilation, btw.

Reply:
@apopiak Is there a way to convert `DispatchResultWithPostInfo` to `DispatchResult` or vice versa? I think using the `?` operator here causes the error that says `the trait `From<DispatchErrorWithPostInfo<PostDispatchInfo>>` is not implemented for `DispatchError`.

Reply:
```suggestion
				pallet_balances::Pallet::<Runtime>::transfer(
					RawOrigin::Signed(caller).into(),
					recipient_account,
					transfer_amount.into(),
				).map(|_| ())?;
```

should work

Reply:
ah, got it wrong, it's 
```suggestion
				pallet_balances::Pallet::<Runtime>::transfer(
					RawOrigin::Signed(caller).into(),
					recipient_account,
					transfer_amount.into(),
				).map_err(|e| e.error)?;
```

Reply:
Yes this works. Thanks